azext_edge/edge/providers/check/base/pod.py

```diff
@@ -121,40 +114,91 @@ def _decorate_pod_condition(condition: bool) -> Tuple[str, str]:
 
         # When pod in obnormal state, sometimes the conditions are not available
         if pod_conditions:
+            conditions_readiness = True
+            conditions_display_list: List[Tuple[str, str]] = []
+            unknown_conditions_display_list: List[Tuple[str, str]] = []
+
+            for condition in pod_conditions:
+                type = condition["type"]
+
+                try:
+                    condition_type = POD_CONDITION_TEXT_MAP[type]
+                    condition_status = condition.get("status") == "True"
+                    conditions_readiness = conditions_readiness and condition_status
+                    pod_condition_deco, status = _decorate_pod_condition(condition=condition_status)
+                except KeyError:
+                    condition_type = type
+                    condition_status = condition.get("status")
+
+                formatted_reason = ""
+                condition_reason = condition.get("reason", "")
+
+                if condition_reason:
+                    # remove the [ and ] to prevent console not printing the text
+                    condition_reason = condition_reason.replace("[", "\\[")
+                    formatted_reason = f"[red]Reason: {condition_reason}[/red]"
+
+                known_condition_values = [value.replace(" ", "").lower() for value in POD_CONDITION_TEXT_MAP.values()]
```

Comment:
can we not trust the pod condition to have spaces correctly?

Reply:
no we can't😣

Reply:
quote your meme
![image](https://github.com/user-attachments/assets/e9f7aeea-5051-4b4c-a4a3-c0fe2abf5d5e)